src/edit-plan-form.ts

```diff
@@ -3,26 +3,63 @@ import { customElement, property } from 'lit/decorators.js';
 
 import type { MonthlyPlan } from './models/plan';
 import './form-sections/cancel';
+import './form-sections/amount';
+import type { MGCEditPlanAmount } from './form-sections/amount';
 
 @customElement('ia-mgc-edit-plan')
 export class IauxEditPlanForm extends LitElement {
   @property({ type: Object }) plan?: MonthlyPlan;
 
   @property({ type: Object }) cancelPlanHandler?: (plan: MonthlyPlan) => void;
 
+  @property({ type: Object }) updateAmountHandler?: (
+    plan: MonthlyPlan,
+    amountUpdates: {
+      amount: number;
+      baseAmount: number;
+      coverFees: boolean;
+      feeCovered: number;
+    }
+  ) => void;
+
+  createRenderRoot() {
+    return this;
+  }
+
+  amountUpdates(status: 'success' | 'fail') {
+    const amountForm = this.querySelector(
+      'ia-mgc-edit-plan-amount'
+    ) as MGCEditPlanAmount;
+    amountForm!.amountUpdated(status);
```

Comment:
How come you're coercing the `amountForm`? It should work or fail gracefully either way.

Reply:
removed, thanks